src/test/java/org/sasanlabs/service/vulnerability/openRedirect/Http3xxStatusCodeBasedInjectionTest.java

```diff
@@ -0,0 +1,658 @@
+        String redirectUrl = "somedomain.com";
+        RequestEntity<String> requestEntity =
+                new RequestEntity<>(
+                        HttpMethod.GET, new URI("https://somedomain.com?returnTo=somedomain.com"));
+        ResponseEntity<?> responseEntity =
+                http3xxStatusCodeBasedInjection.getVulnerablePayloadLevel6(
+                        requestEntity, redirectUrl);
+        assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.FOUND);
+        assertThat(responseEntity.getHeaders().get(LOCATION_HEADER_KEY))
+                .contains("https://somedomain.comsomedomain.com");
+        verify(http3xxStatusCodeBasedInjection, times(1))
+                .getVulnerablePayloadLevel6(requestEntity, "somedomain.com");
+    }
+
+    @Test
+    @DisplayName(
+            "Level 7- test that the returnTo query parameter's value is directly added to the Location header by adding domain as prefix")
+    void
+            test_That_ReturnToQueryParameterValue_IsAddedToLocationHeader_ByAddingDomainToPrefix_Level7()
+                    throws MalformedURLException, URISyntaxException {
+        String redirectUrl = "/somedomain.com";
+        RequestEntity<String> requestEntity =
+                new RequestEntity<>(
+                        HttpMethod.GET, new URI("https://somedomain.com?returnTo=/somedomain.com"));
+        ResponseEntity<?> responseEntity =
+                http3xxStatusCodeBasedInjection.getVulnerablePayloadLevel7(
+                        requestEntity, redirectUrl);
+        assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.FOUND);
+        assertThat(responseEntity.getHeaders().get(LOCATION_HEADER_KEY))
+                .contains("https://somedomain.com/somedomain.com");
+        verify(http3xxStatusCodeBasedInjection, times(1))
+                .getVulnerablePayloadLevel7(requestEntity, "/somedomain.com");
+    }
+
+    @Test
+    @DisplayName(
+            "Level 8- test that the returnTo query parameter's value is not added to the Location header when it is not in the list of whitelisted URLS")
+    void
+            test_That_ReturnToQueryParameterValue_IsNotAddedToLocationHeader_WhenItIsNotInTheListOfWhiteListedURLS_Level8()
+                    throws URISyntaxException {
+        // /somedomain.com is not in the list of whitelisted urls
+        String redirectUrl = "/somedomain.com";
+        RequestEntity<String> requestEntity =
+                new RequestEntity<>(
+                        HttpMethod.GET, new URI("https://somedomain.com?returnTo=/somedomain.com"));
+        ResponseEntity<?> responseEntity =
+                http3xxStatusCodeBasedInjection.getVulnerablePayloadLevel8(
+                        requestEntity, redirectUrl);
+        assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(responseEntity.getHeaders().get(LOCATION_HEADER_KEY)).isEqualTo(null);
+        verify(http3xxStatusCodeBasedInjection, times(1))
```

Comment:
same we have called this method

Reply:
Hi @preetkaran20 ,
I need some clarity, should I remove the verify checks?

Reply:
yes as you have called the method from test case only so checking the verify doesn't give any benefit.